src/NUnitFramework/framework/Constraints/ExactCountConstraint.cs

```diff
@@ -31,9 +31,19 @@ namespace NUnit.Framework.Constraints
     /// item in a collection, succeeding only if a specified
     /// number of items succeed.
     /// </summary>
-    public class ExactCountConstraint : PrefixConstraint
+    public class ExactCountConstraint : Constraint
```

Comment:
I also want to point out that from my perspective it was doing two things that should have been separate before I got here, so perhaps the fix for that should also be isolated from this PR. The change I made was to allow itemConstraint to be null rather than create a dummy `true` constraint. The only reason there are two loops is to save a null check per iteration but I could change it back to a single loop with the logical equivalent of `itemConstraint?.ApplyTo(item).IsSuccess != false` if it makes a big difference right now.